src/OnlineSales/Controllers/ImagesController.cs

```diff
@@ -50,20 +51,34 @@ public async Task<ActionResult> Post([FromForm] ImageCreateDto imageCreateDto)
             using var fileStream = imageCreateDto.Image.OpenReadStream();
             byte[] imageInBytes = new byte[incomingFileSize];
             fileStream.Read(imageInBytes, 0, (int)imageCreateDto.Image.Length);
-
-            Image uploadedImage = new ()
+            
+            var scopeAndFileExists = from i in apiDbContext!.Images!
+                                     where i.ScopeUid == imageCreateDto.ScopeUid.Trim() && i.Name == incomingFileName
+                                     select i;
+            if (scopeAndFileExists.Any())
             {
-                Name = incomingFileName,
-                Size = incomingFileSize,
-                Data = imageInBytes,
-                MimeType = incomingFileMimeType!,
-                ScopeUid = imageCreateDto.ScopeUid,
-                Extension = incomingFileExtension,
-            };
-
-            await apiDbContext.Images!.AddAsync(uploadedImage);
-            await apiDbContext.SaveChangesAsync();
+                Image? uploadedImage = scopeAndFileExists!.FirstOrDefault();
+                uploadedImage!.Data = imageInBytes;
 
+                apiDbContext.Images!.Update(uploadedImage);
+                await apiDbContext.SaveChangesAsync();
```

Comment:
@WASandaruwan, SaveChangesAsync happens twice here and on the line 79. Why not to move it outside of if statement?